### ISSUE TITLE
one set, no burn

### ISSUE THREAD

**Issue by jaredjennings** [jaredjennings/darbrrb#9]
So I finally actually tried to do a real backup with this thing. The backup happened to fit on one set. Nothing got burned! growisofs was not called. I surmise this may always happen for the last set; and when there's one set, the last set is the only set.

**Comment by jaredjennings** [comment]
?? This is tested for, and the tests pass. Did dar not call darbrrb in the way expected?

**Comment by jaredjennings** [comment]
Results inconclusive, pending data reduction. Here's what happened. I put in a debug logging statement to show the whole command line every time darbrrb is run, and reran it. It tried to burn a disc, failed because I didn't have growisofs installed, and dar helpfully offered to re-execute darbrrb. I declined, realized my error, tried to re-execute it myself, but didn't do it right.

Now I have to re-do the whole thing. And it's dar that takes most of the cpu time now, and it's single-threaded. It takes hours. I'm not sure how long because the log messages don't have time stamps. I think I'm getting enough network bandwidth to saturate dar, with my gigabit ethernet and NFSv4. But it's not as easy to replace dar with something simpler as it was with par2: it's the spine of the execution, it does all the hard bits about dealing with weird kinds of files and files getting modified, and I reckon it's likely complicated enough without adding multithreading to it. No word on how much CPU encryption is taking.

This rambly diatribe is going to result in several more issues filed, which should be clearer and better separated.

**Comment by jaredjennings** [comment]
So - here, in this issue:

the data from the log needs to be reduced, the pattern of darbrrb executions checked against the pattern the tests generate, and a specific test made from this pattern, whether that means specializing tests that already exist for this particular case, or writing a new one.

**Comment by jaredjennings** [comment]
state problem, #10.

**Comment by jaredjennings** [comment]
growisofs retry, #11.

**Comment by jaredjennings** [comment]
timestamps, #12.

**Comment by jaredjennings** [comment]
At this point I could call the whole thing a fluke, since the tests obviously test for it already, the sequence of `_create`s from dar is the same as what is tested, I didn't have `growisofs` installed, which the program already says I need when I run it, and I don't have a log file to look back at.

**Comment by jaredjennings** [comment]
!!

I set out to burn this same archive, without encryption, and everything was going fine. I burned two discs of six. Then I realized I didn't want the data bumping around unencrypted, so I quit the whole thing. I made an encrypted one. And *dar never called darbrrb with the last_slice argument!* That's why I got no burn! This seems like a bug in dar, and I'm going to try to duplicate it with a much smaller archive.
